### ISSUE TITLE
named parameter support when destructureing from the constructor

### ISSUE THREAD

**Issue by seebees** [dafny-lang/dafny#2319]
given
```dafny
datatype Foo = Foo(
  nameonly Bar: nat
)
```

I can `var f := Foo( Bar: 5 );`
I can also `var Foo(b) := f;`

I want to be able to use the named parameter.
Something like
`var Foo(b := Bar) := f;` or `var Foo(bar -> b) := f;`

**Comment by MikaelMayer** [comment]
I agree, that would make even possible to use partial matches where you don't need an pattern argument for every parameter.
